lib/daru/dataframe.rb

```diff
@@ -295,11 +295,11 @@ def initialize source, opts={}
           end
         when Hash
           create_vectors_index_with vectors, source
+          vectors_have_same_index = all_vectors_have_equal_indexes?(source)
```

Comment:
I don't quite understand how moving the method above the if condition made it faster. In fact that method will now run even if all the values of the supplied Hash are not `Daru::Vector` objects. Could you explain?

Also, there's a problem here. Since the `all_vectors_have_equal_indexes?` method has been moved outside the `if all_daru_vectors_in_source? source` statement, the `all_vectors_have_equal_indexes?` can get an `Array` in the values of the Hash that is `source`, while it is expecting all the values to be `Daru::Vector` objects. For example:

``` ruby

df = Daru::DataFrame.new({a: [1,2,3], b: [2,3,4]})
```

The code passes because calling `index` on `Array` returns an `Enumerator`. See it [here](https://github.com/inside-track/daru/blob/FIX-DupePerformance/lib/daru/dataframe.rb#L2415).